tests/unit/mixins/route-test.js

```diff
@@ -308,3 +308,70 @@ test('it uses overridden params when loading more data', assert => {
   assert.ok(model.get('reachedInfinity'), 'Should reach infinity');
 
 });
+
+test('it allows overrides/manual invocations of updateInfinityModel', assert => {
+  var RouteObject = Ember.Route.extend(RouteMixin, {
+    model() {
+      return this.infinityModel('item', {perPage: 1});
+    },
+    updateInfinityModel(newObjects) {
+      return this._super(newObjects.setEach('author', 'F. Scott Fitzgerald'));
+    }
+  });
+  var route = RouteObject.create();
+
+  var items = [
+    { id: 1, title: 'The Great Gatsby' },
+    { id: 2, title: 'The Last Tycoon' }
+  ];
+
+  var dummyStore = {
+    find(modelType, findQuery) {
+      var item = items[findQuery.page-1];
+      return new Ember.RSVP.Promise(resolve => {
+        Ember.run(this, resolve, Ember.ArrayProxy.create({
+          content: Ember.A([item]),
+          meta: { total_pages: 2 }
+        }));
+      });
+    }
+  };
+
+  route.store = dummyStore;
+
+  var model;
+  Ember.run(() => {
+    route.model().then(result => {
+      model = result;
+    });
+  });
+
+  var dummyController = Ember.Object.create({
+    model
+  });
+  route.set('controller', dummyController);
+
+  assert.equal(route.get('_canLoadMore'), true);
+  assert.equal(model.get('content.length'), 1);
+
+  Ember.run(() => {
+    route._infinityLoad();
+  });
+
+  assert.equal(route.get('_canLoadMore'), false);
+  assert.equal(model.get('content.length'), 2);
+  assert.equal(model.get('content.lastObject.author'), 'F. Scott Fitzgerald', 'overrides to updateInfinityModel should take effect');
+
+  var newObjects = Ember.ArrayProxy.create({
+    content: Ember.A([
+      { id: 3, title: 'Tender Is the Night' }
+    ])
+  });
```

Comment:
This was a little bit laborious to construct... I might be wrong, but I don't think manual invocations of `updateInfinityModel` as specified in your example @hhff will work even with prototype extensions. However, it's still at least possible, and maybe it's nice to have the flexibility of being able to access meta parameters in `updateInfinityModel`.

So my default plan will be to keep this and update the README so that instead of
```js
var updatedInfinityModel = this.updateInfinityModel([
  { id: 1, name: "Hugh Francis" }
]);
```
we have:
```js
var updatedInfinityModel = this.updateInfinityModel(Ember.ArrayProxy.create({
  content: Ember.A([
    { id: 1, name: "Hugh Francis" }
  ])
});
```
Let me know how that sounds or if I just am misunderstanding prototype extensions!